src/mono/msbuild/android/build/AndroidApp.targets

```diff
@@ -119,8 +119,14 @@
     </ItemGroup>
   </Target>
 
+
+
   <Target Name="_AndroidGenerateAppBundle" DependsOnTargets="_AndroidGenerateRuntimeConfig">
 
+    <PropertyGroup>
+      <RuntimeComponents Condition="'$(RuntimeComponents)' == ''" >marshal-ilgen</RuntimeComponents>
```

Comment:
Would we not want to add `marshal-ilgen` to `RuntimeComponents` if it contained something else? When do we want to include `marshal-ilgen`?

Reply:
Because if RuntimeComponents is already set, we don't want to overwrite it, and delete whatever else some one wanted to include. I tried making RuntimeComponents into an ItemGroup instead, so we could just append it. But that did not work for msbuild reasons I haven't been able to understand yet.

Reply:
We can avoid overwriting what was originally there via
`<RuntimeComponents>$(RuntimeComponents);marshal-ilgen</RuntimeComponents>`

Reply:
I don't think we should use concatenation.
If someone is setting RuntimeComponents explicitly, they should have to specify all of them - including marshal-ilgen if they want it.  (For example testing that we can leave it out will want to leave it out...)

Remember this `RuntimeComponents` property is something we invented in dotnet/runtime and it's not shipping.  Users (ie the mobile and wasm SDKs) set `_MonoComponent` and then look at `_MonoRuntimeComponentLink` or `_MonoRuntimeComponentDontLink` depending on whether they are doing opt-in or opt-out.  So it's only for in-tree testing (using our mobile AppBuilders, or the in-tree wasmApp) that uses this thing.  We can keep it simple to allow testing of weird scenarios.